tests/client/test_search_and_parse_results.py

```diff
@@ -0,0 +1,104 @@
+        b"<search:metrics>\n    <search:query-resolution-time>PT0.007706S</search:query-resolution-time>\n    <search:snippet-resolution-time>PT0.000054S</search:snippet-resolution-time>\n    <search:extract-resolution-time>PT0.002231S</search:extract-resolution-time>\n    <search:total-time>PT0.010856S</search:total-time>\n  </search:metrics>\n</search:response>"  # noqa: E501
+        b"\r\n--foo--\r\n"
+    )
+
+    mock_advanced_search.return_value = response_mock
+
+    # Call perform_advanced_search with mocked api_client and inputs
+    search_results = api_client.search_judgments_and_parse_results(
+        SearchParameters(
+            q="test query",
+            court="test court",
+            judge="test judge",
+            party="test party",
+            neutral_citation="test citation",
+            specific_keyword="test keyword",
+            date_from="2022-01-01",
+            date_to="2022-01-31",
+            page=1,
+            page_size=10,
+        )
+    )
+
+    mock_advanced_search.assert_called_once_with(
+        SearchParameters(
+            q="test query",
+            court="test court",
+            judge="test judge",
+            party="test party",
+            neutral_citation="test citation",
+            specific_keyword="test keyword",
+            order=None,
+            date_from="2022-01-01",
+            date_to="2022-01-31",
+            page=1,
+            page_size=10,
+            show_unpublished=False,
+            only_unpublished=False,
+            collections=["judgment"],
+        )
+    )
+
+    # Assert that SearchResults object was created successfully
+    assert isinstance(search_results, SearchResults)
+    assert search_results.total == "2"
+    assert len(search_results.results) == 2
+    assert search_results.results[0].attrib == {
+        "index": "1",
+        "uri": "/a/b/2011/10.xml",
+        "path": 'fn:doc("/a/b/2011/10.xml")',
+        "score": "40610",
```

Comment:
I'm not sure I like this test. I don't know that we depend on the scope, confidence or fitness, the search result XML is pretty huge and we're broadly testing for "xml in = xml out". 

I'd drop the uncared for attribute tests, and heavily truncate the search results, or consider migrating this out to a test fixture.

Reply:
Yeah you make a good point, that is indeed what Im trying to show here - that xml in = xml out and those particular fields arent special in any way.

I had copied the response (that I am mocking) from an actual call to ML and modified it a bit so that it didnt refer to any real data, but you're right - I can definitely truncate the data a lot - so I'll give that a go!